collections/_article/migrate-animation-from-godot-4-0-to-4-3.md

```diff
@@ -0,0 +1,317 @@
+
+Since `Discrete` assumes that there is no blending, it sets the value of the object during the blending process, but `Continuous` sets the value to the object after the blending process has been completed. This means that if it simply blend tracks with `Discrete` and `Continuous`, the result of the `Continuous` track will always take precedence over the `Discrete` track.
+
+That was a problem because the `Deterministic` blend resetting behavior explained above is done as a `Continuous` mode.
+
+When implementing the `AnimationMixer` and `Deterministic` options in 4.2, we tried to solve that problem by implicitly converting `Discrete` mode to `Continuous` mode when blending occurs.
+
+However, this implicit conversion created another problem when the user understood the behavior of the `Discrete` track and the project relied on the infrequent application of `Discrete` values.
+
+In 4.3, now the user can choose whether or not to force conversions, so they can be handled them explicitly.
+
+**Here is how the options behave when `Discrete` and `Continuous` are blended:**
+
+There are two animations, one is `Continuous` and one is `Discrete`.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/continuous_and_discrete.mp4?1" type="video/mp4">
+</video>
+
+#### Force Continuous
+
+Discrete track is treated as a `Continuous` track and they are blended together.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/force_continuous.mp4?1" type="video/mp4">
+</video>
+
+#### Dominant
+
+Only on frames with `Discrete` track keys, the `Continuous` track results are not reflected as the `Discrete` track values are given priority.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/dominant.mp4?1" type="video/mp4">
+</video>
+
+Note that instead of respecting the Discrete value, flicker may occur during blending.
+
+#### Recessive
+
+In frames with `Discrete` track keys, the `Discrete` track values are potentially reflected, but are immediately overwritten by the `Continuous` track results, so the final result appears to only `Continuous` track.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/recessive.mp4?1" type="video/mp4">
+</video>
+
+Flickering does not occur and tends to look better. But note that if the `Continuous` track is blended even a little, the `Discrete` track result may be hidden.
+
+By the way, note that `AnimationPlayer` set `Recessive` as the default but `AnimationTree` set `Force Continuous` as the default. Since it is good compatibility with `Deterministic` which is enabled by default, and on the basis of the many issues that have been posted from many users in the past about "discrete tracks cannot blend".
```

Comment:
```suggestion
By the way, note that `AnimationPlayer` is set to `Recessive` by default but `AnimationTree` is set to `Force Continuous` by default. Since it works well with the `Deterministic` mode which is enabled by default, and on the basis of the many issues that have been posted from many users who reported that "discrete tracks cannot blend".
```